### PR TITLE
Add GeckoVersion.bump_version_type()

### PR DESCRIPTION
**Pull request by JohanLorenzo**

### CODE REVIEW

File: mozilla_version/gecko.py [hunk 1]
```diff
@@ -412,6 +417,56 @@ def _create_bump_kwargs(self, field):
...
+             * 32.0 is bumped to 32.0esr
+             * 31.0build1 is bumped to 31.0esrbuild1
+             * 31.0build2 is bumped to 31.0esrbuild1
+
+        """
+        try:
+            return self.__class__(**self._create_bump_version_type_kwargs())
+        except (ValueError, PatternNotMatchedError) as e:
+            # TODO Use "raise from" statement once Python 2 support is dropped
+            raise_from(ValueError(
```
Comment (srfraser) [inline mozilla_version/gecko.py:439]: TIL, nice!

File: mozilla_version/test/test_gecko.py [hunk 2]
```diff
@@ -523,6 +523,44 @@ def test_gecko_version_ensures_a_new_added_release_type_is_caught(monkeypatch):
...
+    ('32.0b10', '32.0'),
+    ('32.0b10build3', '32.0build1'),
+
+    ('54.0a1', '54.0a2'),
+    ('54.0a2', '54.0b1'),
+    ('55.0a1', '55.0b1'),
+
+    ('2.0b2', '2.0rc1'),
+    ('2.0rc1', '2.0'),
+    ('2.0rc2', '2.0'),
```
Comment (srfraser) [inline mozilla_version/test/test_gecko.py:547]: Should this test some past 54, so they're not covered by the check against `self._LAST_AURORA_DEVEDITION_AS_VERSION_TYPE`?

File: requirements/docs.in [hunk 3]
```diff
@@ -1,4 +1,4 @@
 # Specifically for generating docs
-Sphinx
+Sphinx<3  # TODO Unpin once https://github.com/sphinx-doc/sphinx/issues/7420 is fixed
```
Comment (srfraser) [inline requirements/docs.in:2]: Woo!

File: mozilla_version/test/test_gecko.py (hunk 2 above)
Comment (JohanLorenzo) [reply mozilla_version/test/test_gecko.py]: There was `('55.0a1', '55.0b1'),` hidden above. I reordered the cases and added some more to be more explicit for future readers 😃

**Pull request merged**